packages/notify-client/src/controllers/engine.ts

```diff
@@ -832,10 +890,27 @@ export class NotifyEngine extends INotifyEngine {
       }
     );
 
-    this.client.lastWatchedAccount.set(LAST_WATCHED_KEY, {
-      [LAST_WATCHED_KEY]: accountId,
+    // Use an array to account for the slim chance to account for
+    // incorrect state where there is more than one account marked as
+    // lastWatched.
+    const currentlastWatchedAccounts = this.client.watchedAccounts
+      .getAll()
+      .filter((account) => account.lastWatched);
+    for (const watchedAccount of currentlastWatchedAccounts) {
```

Comment:
Nit: newline between const assignment and expression block

```suggestion

    for (const watchedAccount of currentlastWatchedAccounts) {
```